extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -119,4 +124,21 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
   {
     inspector.visit("selector", selector);
   }
+
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer, ByteBuffer oldBuffer)
+  {
+    createNewUnion(newBuffer, newPosition, true);
```

Comment:
Would be good to also remove the old union so GC can reclaim the memory. And when a ByteBuffer has no more cached unions, you could remove the NativeMemory and Int2ObjectMap for it.